### DISCUSSION TITLE
Failed to export spans. Agent tries to connect to localhost:4318

### DISCUSSION THREAD

**Discussion by yefisher** [open-telemetry/opentelemetry-java-instrumentation#9861] [Q&A] [answered]
Hey guys, I want to use agent to collect some application data and send it to Grafana.
I use external plugin for gradle to inject agent to the jib and run agent daemon alongside with my app. I use environment variables (e.g. OTEL_EXPORTER_OTLP_ENDPOINT) with my receiver endpoint which ends with 4317.
Application logs show that the agent starts and after few seconds I'm getting next error:
`ERROR [OkHttp http://localhost:4318/...] ThrottlingLogger.java:82 - Failed to export spans. The request could not be executed. Full error message: Failed to connect to localhost/[0:0:0:0:0:0:0:1]:4318`
Is there any known issue or am I doing it in a wrong way?
The version of the agent is 1.31.0

**Comment by laurit** [comment]
It is not a know issue. Most likely you are correct, you are doing something wrong, though it is hard to say what it is. `4318` is the port for `http/protobuf` protocol, while `4317` is for `grpc`, by default agent uses `grpc`.

**Reply by yefisher** [reply]
Most of the examples are quite simple and come down to simply launching the agent and specifying the corresponding environment variables, right? So I do the same using OTEL_EXPORTER_OTLP_LOGS_ENDPOINT, OTEL_EXPORTER_OTLP_TRACES_ENDPOINT and OTEL_EXPORTER_OTLP_METRICS_ENDPOINT environment variables. Logs, traces and metrics are written to the Grafana, but the stacktrace is full of mentioned error.

**Reply by yefisher** [reply]
Sorry for distracting you, logs are not propagated to the grafana, but metrics, traces and spans are good. Mentioned error still there.

**Reply by laurit** [reply, marked as answer]
By default agent does not enable exporting logs (this will change in the 2.0 version of the agent) you have to manually enable it by setting `OTEL_LOGS_EXPORTER=otlp`.
Having spans working is a bit surprising considering the error message you posted says `Failed to export spans`. Unless you provide a minimal application that reproduces the issue it will be hard to tell what causes this.

**Reply by yefisher** [reply]
Hi, 
Thanks a lot, this solution helped me
But mentioned error in the root question still relevant
<img width="1531" alt="image" src="https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/43079080/8103b4cd-01ed-416c-88a4-6d544b45b782">
I have these variables to configure agent
<img width="394" alt="image" src="https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/43079080/0ee2957d-cc45-475f-9d26-cce028119054">

**Reply by laurit** [reply]
I think these logs lines don't come from the otel java agent. Log lines from the agent are formatted slightly differently. Perhaps your application uses a library that configures its own opentelemetry sdk to send traces? For example micrometer tracing that is used in spring boot.

**Reply by yefisher** [reply]
Yeah, you're right!
This might comes from `micrometer-tracing-bridge-otel` 
Thanks a lot for your time and tips.

**Reply by guptamohit24** [reply]
Hello @yefisher 
I'm using spring boot starter integration for open telemetry. I'm also observing these logs while running my app locally. 
Also noticed that it is not coming while I deploy my apps to higher test environment.

Were you able to figure out the cause of this, and what change needed to disable it?

**Reply by jaydeluca** [reply]
@guptamohit24 do you have a collector running locally with a receiver configured for spans?

**Reply by guptamohit24** [reply]
@jaydeluca No, I have just added open telemetry spring boot starter dependency and just running my Spring app. I see these traces on console. However these traces are not coming in dev/sit environments.

**Reply by jaydeluca** [reply]
@guptamohit24 if you are seeing these errors locally, and have no intention of receiving and doing anything with the telemetry emitted, then you can disable the exporters with system properties/environment variables. 

You can find the [docs here](https://opentelemetry.io/docs/languages/java/configuration/#properties-exporters), but essentially you would set these 3:

* `OTEL_LOGS_EXPORTER=none`
* `OTEL_TRACES_EXPORTER=none`
* `OTEL_METRICS_EXPORTER=none`
